server-adapters/resteasy-undertow/src/main/java/org/jboss/resteasy/plugins/server/undertow/UndertowJaxrsServer.java

```diff
@@ -151,11 +154,34 @@ public UndertowJaxrsServer deploy(ResteasyDeployment deployment)
     */
    public UndertowJaxrsServer deploy(ResteasyDeployment deployment, String contextPath)
    {
+      return deploy(deployment, contextPath, null, null);
+   }
+   
+   public UndertowJaxrsServer deploy(ResteasyDeployment deployment, String contextPath, Map<String, String> contextParams, Map<String, String> initParams)
+   {
       if (contextPath == null) contextPath = "/";
       if (!contextPath.startsWith("/")) contextPath = "/" + contextPath;
       DeploymentInfo builder = undertowDeployment(deployment);
       builder.setContextPath(contextPath);
       builder.setDeploymentName("Resteasy" + contextPath);
+      builder.setClassLoader(deployment.getApplication().getClass().getClassLoader());
+      if (contextParams != null)
+      {
+         for (Iterator<String> it = contextParams.keySet().iterator(); it.hasNext(); )
+         {
+            String key = it.next();
+            builder.addInitParameter(key, contextParams.get(key));

```

Comment:
For better performances, the best practice here would be:

for (Entry<String, String> e : contextParams.entrySet()) {
  builder.addInitParameter(e.getKey(), e.getValue());
}

which avoids accessing the map again using the key.